src/context/provider.ts

```diff
@@ -0,0 +1,37 @@
+import { PublishRequest } from "@upstash/qstash";
+import { WorkflowError } from "../error";
+
+/**
+ * copies and updates the request by removing the api field and adding url & headers.
+ *
+ * @param api api field of PublishRequest
+ * @returns updated request
+ */
+export const getProviderInfo = (api: Required<PublishRequest>["api"]) => {
+  if (!api.provider) {
+    throw new WorkflowError("A Provider must be provided.");
+  }
+  if (api.provider.owner === "upstash") {
+    throw new WorkflowError("Upstash provider isn't supported.");
```

Comment:
just curious; is this due to a technical reason or because it's already deprecated in qstash sdk?

Reply:
because it's already deprecated.

I didn't to add this option just to remove it a few months later.